Mathlib/Data/Complex/Basic.lean

```diff
@@ -0,0 +1,1379 @@
+/-
+Copyright (c) 2017 Kevin Buzzard. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kevin Buzzard, Mario Carneiro
+
+! This file was ported from Lean 3 source module data.complex.basic
+! leanprover-community/mathlib commit 92ca63f0fb391a9ca5f22d2409a6080e786d99f7
+! Please do not edit these lines, except to modify the commit id
+! if you have ported upstream changes.
+-/
+import Mathlib.Data.Real.Sqrt
+
+/-!
+# The complex numbers
+
+The complex numbers are modelled as ℝ^2 in the obvious way and it is shown that they form a field
+of characteristic zero. The result that the complex numbers are algebraically closed, see
+`field_theory.algebraic_closure`.
+-/
+
+
+open BigOperators
+
+open Set Function
+
+/-! ### Definition and basic arithmmetic -/
+
+
+/-- Complex numbers consist of two `real`s: a real part `re` and an imaginary part `im`. -/
+structure Complex : Type where
+  re : ℝ
+  im : ℝ
+#align complex Complex
+
+
+-- mathport name: exprℂ
+notation "ℂ" => Complex
+
+namespace Complex
+
+open ComplexConjugate
+
+noncomputable instance : DecidableEq ℂ :=
+  Classical.decEq _
+
+/-- The equivalence between the complex numbers and `ℝ × ℝ`. -/
+@[simps apply]
+def equivRealProd : ℂ ≃ ℝ × ℝ where
+  toFun z := ⟨z.re, z.im⟩
+  invFun p := ⟨p.1, p.2⟩
+  left_inv := fun ⟨_, _⟩ => rfl
+  right_inv := fun ⟨_, _⟩ => rfl
+#align complex.equiv_real_prod Complex.equivRealProd
+
+@[simp]
+theorem eta : ∀ z : ℂ, Complex.mk z.re z.im = z
+  | ⟨_, _⟩ => rfl
+#align complex.eta Complex.eta
+
+@[ext]
+theorem ext : ∀ {z w : ℂ}, z.re = w.re → z.im = w.im → z = w
+  | ⟨_, _⟩, ⟨_, _⟩, rfl, rfl => rfl
+#align complex.ext Complex.ext
+
+theorem ext_iff {z w : ℂ} : z = w ↔ z.re = w.re ∧ z.im = w.im :=
+  ⟨fun H => by simp [H], fun h => ext h.1 h.2⟩
+#align complex.ext_iff Complex.ext_iff
+
+theorem re_surjective : Surjective re := fun x => ⟨⟨x, 0⟩, rfl⟩
+#align complex.re_surjective Complex.re_surjective
+
+theorem im_surjective : Surjective im := fun y => ⟨⟨0, y⟩, rfl⟩
+#align complex.im_surjective Complex.im_surjective
+
+@[simp]
+theorem range_re : range re = univ :=
+  re_surjective.range_eq
+#align complex.range_re Complex.range_re
+
+@[simp]
+theorem range_im : range im = univ :=
+  im_surjective.range_eq
+#align complex.range_im Complex.range_im
+
+-- Porting note: refactored instance to allow `norm_cast` to work
+@[coe]
+def ofReal' (r : ℝ) : ℂ :=
```

Comment:
Why can't this be called `ofReal` without the `'`?
Also, please rename all occurences of `of_real` below.

Reply:
There is another function called `ofReal` based on porting which is a homomorphism.
I will fix the `of_real` names.

Reply:
Fixed the `of_real` in names.

Reply:
Can you add a docstring that explains this naming?